sim/core/dot.go

```diff
@@ -124,12 +125,29 @@ func (dot *Dot) RescheduleNextTick(sim *Simulation) {
 	sim.AddPendingAction(dot.tickAction)
 }
 
+// Snapshots and activates the Dot
+// If the Dot is running it's duration will be refreshed and
+// if there was a next Dot happening this will carry over to the new Dot
 func (dot *Dot) Apply(sim *Simulation) {
+
 	dot.TakeSnapshot(sim, false)
 
-	dot.Cancel(sim)
 	dot.TickCount = 0
-	dot.RecomputeAuraDuration()
+
+	// we a have running dot tick
+	// the next tick never get's clipped and is added onto the dot's time for hasted dots
+	// see: https://github.com/wowsims/cata/issues/50
+	if dot.tickAction != nil {
```

Comment:
`dot.tickAction != nil && !dot.tickAction.canceled`

Its safer to check if the action was canceled as well as not nil